### PR TITLE
Add exception handling and fix minor issues

### PR DESCRIPTION
**Pull request by nghiant2710**

### CODE REVIEW

File: application.py [hunk 1]
```diff
@@ -9,4 +9,15 @@ def restart(self, app_id):
         return self.request('/application/' + str(app_id) + '/restart', 'POST')
 
     def get_by_id(self, app_id):
-        return self.request('application?id=' + str(app_id), 'GET')[0]
+    	params = {
```
Comment (dfunckt) [inline application.py]: The whole block is badly indented here

**Comment by dfunckt** [comment]
Why so many print statements?

Btw, prefer to use the print() function instead, it's python 3 compatible.

File: resources.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#
+#    This is a python library for resources like message templates etc.
+#
+
+class Message:
```
Comment (dfunckt) [inline resources.py]: It's best to always extend from object, ie:

Class Message(object):
  # ...

Old style classes are deprecated and I think don't even work on Python 2.7

File: device.py [hunk 3]
```diff
@@ -27,14 +36,21 @@ def get_by_name(self, name):
...
+            # unexpected exception
+            pass
 
     def identify(self, uuid):
         data = {
             'uuid': uuid
         }
         return self.request('/blink', 'POST', data=data)
-
+
```
Comment (dfunckt) [inline device.py]: Stray whitespace here.

File: device.py [hunk 4]
```diff
@@ -27,14 +36,21 @@ def get_by_name(self, name):
...
         }
-        return self.request('device', 'GET', params=params)[0]
+        try:
+            return self.request('device', 'GET', params=params)
+        except IndexError:
+            # found no device
+            print Message.NO_DEVICE_FOUND.format(value=uuid,dev_att="name")
+        except:
+            # unexpected exception
+            pass
```
Comment (dfunckt) [inline device.py]: Shouldn't this be somehow handled here?

File: device.py [hunk 5]
```diff
@@ -1,4 +1,6 @@
 from .baseapi import BaseAPI
+from resources import Message
+import sys
```
Comment (dfunckt) [inline device.py]: I personally find sorting my imports sensibly helps reviewing my code.

I follow this rules:

- First python imports
- then third party libraries
- then project/package imports
- then local imports

All "sections" are kept sorted by name and separated by a blank line. I also put relative imports below absolute ones.

**Comment by dfunckt** [comment]
I find some methods are not consistent in returning things -- you should make sure return values make sense and are not a dict, or an int or None, etc, depending on conditions.

**Comment by nghiant2710** [comment]
@dfunckt I've updated your comments.
About returning things, some methods are not required a return value, like API call to blink a device. So I just leave it with a return to break from that function.

File: device.py [hunk 6]
```diff
@@ -1,5 +1,7 @@
-from .baseapi import BaseAPI
+import sys
 
+from .baseapi import BaseAPI
+from .resources import Message
```
Comment (dfunckt) [inline device.py]: Ah, that's better :)

File: resources.py [hunk 7]
```diff
@@ -0,0 +1,7 @@
+#
+#    This is a python library for resources like message templates etc.
+#
```
Comment (dfunckt) [inline resources.py]: Minor thing: prefer to document Python code in triple-quote blocks. For a module "docstring" I would write it to the top like this:

```
  """
  This module is doing this and that.
  """
```

Similarily for function, class or method, I'd codument it just below the signature, ie:

```
class Foo(object):
  """
  Foo blah..
  """
  def bar(self):
    """
    This is a method docstring
    """
```

Python (and third-party tools) does a great job taking those "docstrings" and generating documentation.

File: resources.py [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+
+class Message(object):
+	NO_DEVICE_FOUND="There is no device with {dev_att}: {value}! Please check the device {dev_att}!"
+	NO_APPLICATION_FOUND="There is no application with {app_att}: {value}! Please check the application {app_att}!"
```
Comment (dfunckt) [inline resources.py]: Also minor, there should be a space around the assignment "=" there.

**Comment by dfunckt** [comment]
Looks good overall :) If you like, please address a couple of new comments I left, but they're minor things considering this is a work in progress.

Re: return values, my comment may not apply to this PR, given that I read it out of context (ie. this diff). It still holds though as a general suggestion/best practice.

:)

File: application.py [hunk 9]
```diff
@@ -9,4 +9,15 @@ def restart(self, app_id):
...
+            'eq': app_id
+        }
+        try:
+            return self.request('application', 'GET', params=params)
+        except IndexError:
+            # found no application
+            print(Message.NO_APPLICATION_FOUND.format(value=app_id,dev_att="id"))
+        except:
+            # unexpected exception
+            pass
```
Comment (dfunckt) [inline application.py]: You still don't handle the exceptions -- same in ``device.py`` -- you just silence them. I think users of the library would like to know that an error has happened.

Comment (nghiant2710) [inline application.py]: @dfunckt . My bad. I didn't notice that. I was testing all the methods so wanted to ignore unexpected errors.

**Pull request merged**